include/grpc/grpc_security.h

```diff
@@ -808,49 +808,81 @@ GRPCAPI int grpc_tls_key_materials_config_set_key_materials(
     credential reload request.
-    - cb and cb_user_data represent a gRPC-provided
-      callback and an argument passed to it.
-    - key_materials_config is an in/output parameter containing currently
-      used/newly reloaded credentials. If credential reload does not result in
-      a new credential, key_materials_config should not be modified. The same
-      key_materials_config object can be updated if new key materials is
-      available.
-    - status and error_details are used to hold information about
-      errors occurred when a credential reload request is scheduled/cancelled.
-    - config is a pointer to the unique grpc_tls_credential_reload_config
+    Note that we currently only support synchronous credential reloading. All
+    the fields related to asynchronous credential reloading can be ignored for
+    now.
+    - cb: a callback-function provided by gRPC stack for users to indicate an
+      asynchronous reloading operation is complete. Users are expected to call
+      this function when asynchronous reloading is done.
+      This field is only applicable in the asynchronous mode which is currently
+      not implemented.
+    - cb_user_data: a pointer used to retrieve some implementations in C core.
+      This is needed because after users call |cb|, we will use this field to
+      notify the gRPC stack to continue the handshake.
+      Users are not expected to interact with this field directly.
+    - key_materials_config: an in/output parameter containing currently
+      used/newly reloaded credentials. This field is provided by the gRPC stack,
+      and users can use it to handle credential reloading.
+      Note that, key_materials_config is not a per-connection parameter, which
+      means it might be shared by multiple connections.
+    - status: a per-connection parameter to indicate the finish status of a
+      schedule/cancel request. Users are expected to manage this field properly
+      in order to do credential reloading.
+      If set to |GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_NEW|, gRPC stack would
+      use whatever users set in |key_materials_config| as the new credential.
+    - err_details: a per-connection parameter to hold the error message if a
+      schedule/cancel request fails.
+      This field is provided by the gRPC stack, and users can use it to handle
+      failure cases.
+    - config: a pointer to the unique |grpc_tls_credential_reload_config|
       instance that this argument corresponds to.
-    - context is a pointer to a wrapped language implementation of this
-      grpc_tls_credential_reload_arg instance.
-    - destroy_context is a pointer to a caller-provided method that cleans
-      up any data associated with the context pointer.
-    It is used for experimental purposes for now and subject to change.
+      This is needed because when C core stack invoke APIs in C core configs,
+      it will eventually need to invoke wrap language config APIs implemented by
```

Comment:
This explanation is correct, but can we find a better place for it, instead of specifying it in `grpc_security.h`? I do not think users need to know about it. Another idea that came to my mind is that, how about we specify the comments in grpc_security.h only targeting grpc users (i.e., implement these configs), and put these implementation details in some source and other header files? Also, I think it may be worthwhile to specify which fields of this arg users need to touch (i.e., can modify) and which ones should be kept still. HDYT?